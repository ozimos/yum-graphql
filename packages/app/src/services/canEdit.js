import differenceInMinutes from 'date-fns/difference_in_minutes';

export default (updatedAt) => {
  const orderEditMinutes = parseInt(process.env.ORDER_EDIT_MINUTES, 10) || 15;
  const minutesSinceOrder = differenceInMinutes(new Date(), updatedAt);
  return orderEditMinutes - minutesSinceOrder > 0;
};
