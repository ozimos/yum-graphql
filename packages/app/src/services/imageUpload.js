import axios from 'axios';

const imageUpload = (files, callback) => {

  const formData = new FormData();
  formData.append('file', files[0]);
  formData.append('tags', 'meals');
  formData.append('upload_preset', process.env.CLOUDINARY_UPLOAD_PRESET);
  formData.append('api_key', process.env.CLOUDINARY_API_KEY);
  formData.append('timestamp', (Date.now() / 1000) || 0);

  const config = {

    onUploadProgress(ProgressEvent) {
      let percentCompleted = 0;

      const totalLength = ProgressEvent.lengthComputable
        ? ProgressEvent.total
        : ProgressEvent.target
          .getResponseHeader('content-length')
        || ProgressEvent.target
          .getResponseHeader('x-decompressed-content-length');

      if (totalLength !== null) {
        percentCompleted = Math
          .round((ProgressEvent.loaded * 100) / totalLength);
      }

      callback(percentCompleted);
    }
  };

  return axios.post(process.env.CLOUDINARY_URL, formData, {
    headers: { 'X-Requested-With': 'XMLHttpRequest' }, ...config
  });
};
export default imageUpload;
