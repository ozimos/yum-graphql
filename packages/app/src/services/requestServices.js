import axios from 'axios';
import authHeader from './authHeader';

const requestServices = (url, requestMethod, data) => {
  const requestOptions = {
    url,
    method: requestMethod || 'get',
    headers: {
      'Content-Type': 'application/json',
      Accept: 'application/json',
      ...authHeader()
    }
  };

  if (data) {
    requestOptions.data = data;
  }
  return axios(requestOptions);
};

export default requestServices;
