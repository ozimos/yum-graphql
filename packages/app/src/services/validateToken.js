import jwt from 'jsonwebtoken';

const validateToken = (token) => {
  let valid = true;
  const decoded = jwt.decode(token);

  const { exp, isCaterer, userId, firstName } = decoded;

  if (exp < Math.floor(Date.now() / 1000)) {
    valid = false;
  }

  return { valid, isCaterer, userId, firstName };
};

export default validateToken;
