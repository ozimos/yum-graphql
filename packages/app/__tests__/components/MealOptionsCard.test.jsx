/* global React:false, shallow:false mount:false toJson:false */

import { MealOptionsCard, mapStateToProps } from
  '../../src/components/mealCard/ConnectedMealOptionsCard';
import * as imageUpload from '../../src/services/imageUpload';
import mockStore from '../__mocks__/mockStore';

import { meal } from '../__mocks__/mealDataMock';

const props = {
  ...meal.data,
  dispatch: jest.fn(),
};

const store = mockStore({
  connecting: true
});

describe('MealOptionsCard Component', () => {

  const shallowSetup = () => shallow(<MealOptionsCard {...props} />);
  const mountSetup = () => mount(<MealOptionsCard {...props} store={store} />);

  it('should render without throwing an error', () => {
    const wrapper = shallowSetup();

    expect(wrapper
      .exists(<div className="card" />)).toBe(true);
  });

  it('renders shallow component correctly', () => {
    const wrapper = shallowSetup();

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders mounted component correctly', () => {
    const wrapper = mountSetup();
    wrapper.setState({ canSubmit: false });
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should call `setUploadPercent` when an image is selected' +
   ' for upload', () => {
    const wrapper = shallowSetup();

    const setUploadPercentSpy = jest.spyOn(
      wrapper.instance(),
      'setUploadPercent'
    );
    wrapper.instance().setUploadPercent(20);
    expect(setUploadPercentSpy).toHaveBeenCalled();
  });

  it('should call `handleDrop` when the upload image button is clicked', () => {
    const imageUploadSpy = jest.spyOn(imageUpload, 'default');
    imageUploadSpy.mockResolvedValue({ data: { secure_url: 'url' } });
    const wrapper = shallowSetup();

    const handleDropSpy = jest.spyOn(wrapper.instance(), 'handleDrop');
    wrapper.instance().handleDrop([]);
    expect(handleDropSpy).toHaveBeenCalled();
  });

  it(
    'should call `openMealEditorModal` when the edit button is clicked',
    () => {
      const wrapper = shallowSetup();

      const openMealEditorModalSpy = jest.spyOn(
        wrapper.instance(),
        'openMealEditorModal'
      );
      wrapper.instance().openMealEditorModal();
      expect(openMealEditorModalSpy).toHaveBeenCalled();
    }
  );

  it(
    'should call `closeMealEditorModal` when the close button is clicked',
    () => {
      const wrapper = shallowSetup();

      const closeMealEditorModalSpy = jest.spyOn(
        wrapper.instance(),
        'closeMealEditorModal'
      );
      wrapper.instance().closeMealEditorModal();
      expect(closeMealEditorModalSpy).toHaveBeenCalled();
    }
  );

  it(
    'should call `openDeleteConfirmModal` when the delete button is clicked',
    () => {
      const wrapper = shallowSetup();

      const openDeleteConfirmModalSpy = jest.spyOn(
        wrapper.instance(),
        'openDeleteConfirmModal'
      );
      wrapper.instance().openDeleteConfirmModal();
      expect(openDeleteConfirmModalSpy).toHaveBeenCalled();
    }
  );

  it(
    'should call `closeDeleteConfirmModal` when the close button is clicked',
    () => {
      const wrapper = shallowSetup();

      const closeDeleteConfirmModalSpy = jest.spyOn(
        wrapper.instance(),
        'closeDeleteConfirmModal'
      );
      wrapper.instance().closeDeleteConfirmModal();
      expect(closeDeleteConfirmModalSpy).toHaveBeenCalled();
    }
  );

  it(
    'should call `deleteMeal` when the confirm delete button is clicked',
    () => {
      const wrapper = shallowSetup();

      const deleteMealSpy = jest.spyOn(
        wrapper.instance(),
        'deleteMeal'
      );
      wrapper.instance().deleteMeal('abc');
      expect(deleteMealSpy).toHaveBeenCalled();
    }
  );

  it(
    'should call `handleUpdateMeal` when the continue button is clicked',
    () => {
      const wrapper = shallowSetup();

      const handleUpdateMealSpy = jest.spyOn(
        wrapper.instance(),
        'handleUpdateMeal'
      );
      wrapper.instance().handleUpdateMeal(meal.data, 'abc');
      expect(handleUpdateMealSpy).toHaveBeenCalled();
    }
  );

  it('should call `disableButton` when the form input data is wrong', () => {
    const wrapper = shallowSetup();

    const disableButtonSpy = jest.spyOn(
      wrapper.instance(),
      'disableButton'
    );
    wrapper.instance().disableButton();
    expect(disableButtonSpy).toHaveBeenCalled();
  });

  it('should call `enableButton`when the form input data is correct', () => {
    const wrapper = shallowSetup();

    const enableButtonSpy = jest.spyOn(
      wrapper.instance(),
      'enableButton'
    );
    wrapper.instance().enableButton();
    expect(enableButtonSpy).toHaveBeenCalled();
  });

  it('should call delete meal on click of delete button', () => {
    const wrapper = shallowSetup();

    const deleteMealSpy = jest.spyOn(
      wrapper.instance(),
      'deleteMeal'
    );
    const closeDeleteConfirmModalSpy = jest.spyOn(
      wrapper.instance(),
      'closeDeleteConfirmModal'
    );
    wrapper.find('.btn.delete').simulate('click');
    expect(closeDeleteConfirmModalSpy).toHaveBeenCalled();
    expect(deleteMealSpy).toHaveBeenCalled();
  });

  it('should map state to props', () => {
    const initialState = {
      dispatch: jest.fn(),
      mealsReducer: {
        connecting: false,
        mealError: ''
      }
    };

    const tree = mapStateToProps(initialState);
    expect(tree).toMatchSnapshot();
  });
});

