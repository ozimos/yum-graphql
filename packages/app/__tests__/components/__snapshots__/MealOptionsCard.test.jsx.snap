// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MealOptionsCard Component renders mounted component correctly 1`] = `
<MealOptionsCard
  connecting={false}
  dispatch={[MockFunction]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <div
    className="card"
  >
    <div
      className="overlay-container"
    >
      <div
        className="overlay"
      >
        <div
          className="flexbox"
        >
          <span
            className="long_string text_left"
          />
          <span>
            ₦
          </span>
        </div>
      </div>
      <img
        alt="Meal"
        className="fluid-img"
      />
    </div>
    <p
      className="descrip long_string"
    />
    <button
      className="btn pad-btn modal-open"
      onClick={[Function]}
    >
      Edit
    </button>
    <button
      className="title-button card-btn"
      onClick={[Function]}
    >
      ✖
    </button>
  </div>
  <Modal
    ariaHideApp={true}
    bodyOpenClassName="ReactModal__Body--open"
    className="modal-content"
    closeTimeoutMS={0}
    contentLabel="Input Modal"
    isOpen={false}
    onRequestClose={[Function]}
    parentSelector={[Function]}
    portalClassName="ReactModalPortal"
    shouldCloseOnEsc={true}
    shouldCloseOnOverlayClick={true}
    shouldFocusAfterRender={true}
    shouldReturnFocusAfterClose={true}
  >
    <ModalPortal
      ariaHideApp={true}
      bodyOpenClassName="ReactModal__Body--open"
      className="modal-content"
      closeTimeoutMS={0}
      contentLabel="Input Modal"
      defaultStyles={
        Object {
          "content": Object {
            "WebkitOverflowScrolling": "touch",
            "background": "#fff",
            "border": "1px solid #ccc",
            "borderRadius": "4px",
            "bottom": "40px",
            "left": "40px",
            "outline": "none",
            "overflow": "auto",
            "padding": "20px",
            "position": "absolute",
            "right": "40px",
            "top": "40px",
          },
          "overlay": Object {
            "backgroundColor": "rgba(255, 255, 255, 0.75)",
            "bottom": 0,
            "left": 0,
            "position": "fixed",
            "right": 0,
            "top": 0,
          },
        }
      }
      isOpen={false}
      onRequestClose={[Function]}
      parentSelector={[Function]}
      portalClassName="ReactModalPortal"
      shouldCloseOnEsc={true}
      shouldCloseOnOverlayClick={true}
      shouldFocusAfterRender={true}
      shouldReturnFocusAfterClose={true}
      style={
        Object {
          "content": Object {},
          "overlay": Object {},
        }
      }
    />
  </Modal>
  <Modal
    ariaHideApp={true}
    bodyOpenClassName="ReactModal__Body--open"
    className="modal-content"
    closeTimeoutMS={0}
    contentLabel="Confirmation Modal"
    isOpen={false}
    onRequestClose={[Function]}
    parentSelector={[Function]}
    portalClassName="ReactModalPortal"
    shouldCloseOnEsc={true}
    shouldCloseOnOverlayClick={false}
    shouldFocusAfterRender={true}
    shouldReturnFocusAfterClose={true}
  >
    <ModalPortal
      ariaHideApp={true}
      bodyOpenClassName="ReactModal__Body--open"
      className="modal-content"
      closeTimeoutMS={0}
      contentLabel="Confirmation Modal"
      defaultStyles={
        Object {
          "content": Object {
            "WebkitOverflowScrolling": "touch",
            "background": "#fff",
            "border": "1px solid #ccc",
            "borderRadius": "4px",
            "bottom": "40px",
            "left": "40px",
            "outline": "none",
            "overflow": "auto",
            "padding": "20px",
            "position": "absolute",
            "right": "40px",
            "top": "40px",
          },
          "overlay": Object {
            "backgroundColor": "rgba(255, 255, 255, 0.75)",
            "bottom": 0,
            "left": 0,
            "position": "fixed",
            "right": 0,
            "top": 0,
          },
        }
      }
      isOpen={false}
      onRequestClose={[Function]}
      parentSelector={[Function]}
      portalClassName="ReactModalPortal"
      shouldCloseOnEsc={true}
      shouldCloseOnOverlayClick={false}
      shouldFocusAfterRender={true}
      shouldReturnFocusAfterClose={true}
      style={
        Object {
          "content": Object {},
          "overlay": Object {},
        }
      }
    />
  </Modal>
</MealOptionsCard>
`;

exports[`MealOptionsCard Component renders shallow component correctly 1`] = `
<React.Fragment>
  <div
    className="card"
  >
    <div
      className="overlay-container"
    >
      <div
        className="overlay"
      >
        <div
          className="flexbox"
        >
          <span
            className="long_string text_left"
          />
          <span>
            ₦
          </span>
        </div>
      </div>
      <img
        alt="Meal"
        className="fluid-img"
      />
    </div>
    <p
      className="descrip long_string"
    />
    <button
      className="btn pad-btn modal-open"
      onClick={[Function]}
    >
      Edit
    </button>
    <button
      className="title-button card-btn"
      onClick={[Function]}
    >
      ✖
    </button>
  </div>
  <Modal
    ariaHideApp={true}
    bodyOpenClassName="ReactModal__Body--open"
    className="modal-content"
    closeTimeoutMS={0}
    contentLabel="Input Modal"
    isOpen={false}
    onRequestClose={[Function]}
    parentSelector={[Function]}
    portalClassName="ReactModalPortal"
    shouldCloseOnEsc={true}
    shouldCloseOnOverlayClick={true}
    shouldFocusAfterRender={true}
    shouldReturnFocusAfterClose={true}
  >
    <div
      className="title flexbox navbar-fixed"
    >
      <h4
        className="shrink"
      >
        Meal Editor
      </h4>
      <button
        className="btn title-button"
        onClick={[Function]}
      >
        ✖
      </button>
    </div>
    <div
      className="form3-grid"
    >
      <div>
        <Formsy
          className="form3"
          disabled={false}
          getErrorMessage={[Function]}
          getErrorMessages={[Function]}
          getValue={[Function]}
          hasValue={[Function]}
          isFormDisabled={[Function]}
          isFormSubmitted={[Function]}
          isPristine={[Function]}
          isRequired={[Function]}
          isValid={[Function]}
          isValidValue={[Function]}
          mapping={null}
          onChange={[Function]}
          onError={[Function]}
          onInvalid={[Function]}
          onInvalidSubmit={[Function]}
          onReset={[Function]}
          onSubmit={[Function]}
          onValid={[Function]}
          onValidSubmit={[Function]}
          preventExternalInvalidation={false}
          resetValue={[Function]}
          setValidations={[Function]}
          setValue={[Function]}
          showError={[Function]}
          showRequired={[Function]}
          validationErrors={null}
        >
          <Formsy(Input)
            innerRef={[Function]}
            name="title"
            placeholder="Meal Title"
            required={false}
            typeOfInput="text"
            validationError="Please enter the meal title"
            validationErrors={
              Object {
                "maxLength": "input must be shorter than 50 characters",
                "minLength": "input must be longer than 1 character",
              }
            }
            validations={
              Object {
                "maxLength": 48,
                "minLength": 1,
              }
            }
          />
          <Formsy(Input)
            innerRef={[Function]}
            name="price"
            placeholder="₦undefined"
            required={false}
            typeOfInput="number"
            validationError="Please enter a price"
            validationErrors={
              Object {
                "isOnlyInt": "price must be integer",
                "maxLength": "input must be shorter than 50 characters",
                "minLength": "input must be longer than 1 character",
              }
            }
            validations={
              Object {
                "isOnlyInt": [Function],
                "maxLength": 48,
                "minLength": 1,
              }
            }
          />
          <Formsy(TextArea)
            innerRef={[Function]}
            name="description"
            placeholder="Description"
            required={false}
            validationError=""
            validationErrors={
              Object {
                "maxLength": "input must be shorter than 100 characters",
                "minLength": "input must be longer than 5 characters",
              }
            }
            validations={
              Object {
                "maxLength": 98,
                "minLength": 5,
              }
            }
          />
          <Formsy(Input)
            innerRef={[Function]}
            name="imageUrl"
            required={false}
            style={
              Object {
                "display": "none",
              }
            }
            typeOfInput="url"
            validationError="Please select an image"
            validationErrors={
              Object {
                "isUrl": "A valid url was not supplied",
                "maxLength": "input must be shorter than 100 characters",
                "minLength": "input must be longer than 5 characters",
              }
            }
            validations={
              Object {
                "isUrl": true,
                "maxLength": 98,
                "minLength": 5,
              }
            }
          />
        </Formsy>
      </div>
      <div
        className="overlay-container"
      >
        <div
          className="overlay full"
        >
          <t
            accept="image/*"
            className="dropzone"
            disableClick={false}
            disablePreview={false}
            disabled={false}
            maxSize={Infinity}
            minSize={0}
            multiple={true}
            onDrop={[Function]}
            preventDropOnDocument={true}
          >
            <button
              className="btn"
            >
               Select an Image
            </button>
          </t>
        </div>
        <div
          id="meal_image"
        >
          <img
            alt="meal"
            className="fluid-img"
          />
        </div>
      </div>
      <button
        className="btn btn-disabled"
        disabled={true}
        onClick={[Function]}
        type="submit"
      >
        <p>
          Continue
        </p>
      </button>
    </div>
  </Modal>
  <Modal
    ariaHideApp={true}
    bodyOpenClassName="ReactModal__Body--open"
    className="modal-content"
    closeTimeoutMS={0}
    contentLabel="Confirmation Modal"
    isOpen={false}
    onRequestClose={[Function]}
    parentSelector={[Function]}
    portalClassName="ReactModalPortal"
    shouldCloseOnEsc={true}
    shouldCloseOnOverlayClick={false}
    shouldFocusAfterRender={true}
    shouldReturnFocusAfterClose={true}
  >
    <p>
      This action will delete your meal. Do you want to continue
    </p>
    <div
      className="title flexbox"
    >
      <button
        className="btn delete"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "red !important",
          }
        }
      >
        <p>
          Yes
        </p>
      </button>
      <button
        className="btn"
        onClick={[Function]}
      >
        <p>
          No
        </p>
      </button>
    </div>
  </Modal>
</React.Fragment>
`;

exports[`MealOptionsCard Component should map state to props 1`] = `
Object {
  "connecting": false,
  "mealError": "",
}
`;
